src/app/login/login.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder } from '@angular/forms';
+import { first } from 'rxjs';
+import { AuthService } from '../Services/auth.service';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.css']
+})
+export class LoginComponent implements OnInit {
+
+  
+  constructor(
+    private fb:FormBuilder,
+    private authService:AuthService,
+    private router:Router
+  ) { }
+
+
+  errorMessage:Boolean=false
+
+  loginForm=this.fb.group({
+    username:[''],
+    password:['']
+  })
+  ngOnInit(): void {
+  }
+
+  login():void{
+    this.authService.login(this.loginForm.value['username']!,this.loginForm.value['password']!).pipe(first())
```

Comment:
**suggestion:** you should try to have this variables in some constants before passing them to the login method, as to be more readable for the people